### PR TITLE
Fix for leap year schedules.

### PR DESCRIPTION
**Pull request by joseph-robertson**

### CODE REVIEW

**Comment by joseph-robertson** [comment]
The forward translator has some problem with 2/29 if the epw is not a leap year.

File: measures/ERIHotWaterAndAppliances/resources/weather.rb [hunk 1]
```diff
@@ -75,8 +75,14 @@ def self.actual_timestamps(model, runner, measure_dir)
       end
       return actual_timestamps
     end
-    return nil
+    return []
```
Comment (shorowit) [inline measures/ERIHotWaterAndAppliances/resources/weather.rb]: Why did you change this? It looks like the [timeseries measure is expecting nil or an array](https://github.com/NREL/OpenStudio-BEopt/blob/master/measures/TimeseriesCSVExport/measure.rb#L304).

Comment (shorowit) [reply measures/ERIHotWaterAndAppliances/resources/weather.rb]: Nevermind, I see now that you changed that line in this commit.

File: measures/ResidentialLocation/measure.rb [hunk 2]
```diff
@@ -151,7 +151,9 @@ def run(model, runner, user_arguments)
     # ----------------
     yd = model.getYearDescription
     if epw_file.startDateActualYear.is_initialized
-      yd.setIsLeapYear(true) # TODO: set this if?
+			if weather.has_leap_day(model, runner, File.dirname(__FILE__))
+				yd.setIsLeapYear(true)
+			end
```
Comment (shorowit) [inline measures/ResidentialLocation/measure.rb]: Please no tabs.

Comment (shorowit) [reply measures/ResidentialLocation/measure.rb]: I don't think you should need to do this. It appears that you should simply be able to use `yd.isLeapYear` if you set the calendar year first? See https://github.com/NREL/OpenStudio/blob/72e4961a2d6547dcb40be48f722345b5691e3a74/openstudiocore/src/model/test/YearDescription_GTest.cpp#L52-L58

File: measures/ResidentialLocation/measure.rb [hunk 3]
```diff
@@ -132,13 +131,13 @@ def run(model, runner, user_arguments)
     # Set mains water temperatures
     # ----------------------------
     
-    avgOAT = UnitConversions.convert(weather.data.AnnualAvgDrybulb,"F","C")
+    avgOAT = OpenStudio::convert(weather.data.AnnualAvgDrybulb,"F","C").get
```
Comment (shorowit) [inline measures/ResidentialLocation/measure.rb]: Some lines in this file got reverted.

File: measures/ResidentialLocation/measure.rb [hunk 4]
```diff
@@ -3,7 +3,6 @@
 
 require "#{File.dirname(__FILE__)}/resources/weather"
 require "#{File.dirname(__FILE__)}/resources/constants"
-require "#{File.dirname(__FILE__)}/resources/unit_conversions"
```
Comment (shorowit) [inline measures/ResidentialLocation/measure.rb]: This line needs to be there.

File: resources/schedules.rb [hunk 5]
```diff
@@ -1,5 +1,4 @@
 # TODO: Need to handle vacations
-require "#{File.dirname(__FILE__)}/unit_conversions"
```
Comment (shorowit) [inline resources/schedules.rb]: This line too.
